// Версия синтаксиса прото-файла
syntax = "proto3";

// Пространство имён (package), используется для избежания конфликтов имён
package autotests;

// Определение gRPC сервиса CourseService
service CourseService {
  // Метод GetCourse принимает GetCourseRequest и возвращает GetCourseResponse
  rpc GetCourse (GetCourseRequest) returns (GetCourseResponse);
}

// Сообщение запроса — содержит только идентификатор курса
message GetCourseRequest {
  string course_id = 1; // Поле с порядковым номером 1
}

// Сообщение ответа — содержит информацию о курсе
message GetCourseResponse {
  // course_id (строка) — идентификатор курса.
  // title (строка) — название курса.
  // description (строка) — описание курса.
  string course_id = 1;
  string title = 2;
  string description = 3;

}